src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -869,7 +868,15 @@ private void WriteCompileTimeAssemblies()
             {
                 WriteItems(
                     _compileTimeTarget,
-                    package => package.CompileTimeAssemblies);
+                    package => package.CompileTimeAssemblies,
```

Comment:
That is the only product change? I thought that is only a cache

Reply:
We always read from the cache to make sure the cached and non-cached behavior are consistent.  So writing the metadata into the cache should be all that is necessary.